File: frontend/Business/tg_business_bot/Models/Order.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from enum import Enum

from Models.OrderItem import OrderItem
from Models.Client import Client

class OrderStatus(Enum):
    IN_ASSEMBLY = "InAssembly"
    COOKING = "Cooking"
    DONE = "Done"


class Order(BaseModel):
    id: Optional[int]
    client_id: int
    table_id: int
    order_items: List[OrderItem]
    comment: Optional[str]
    date_time: datetime
    status: OrderStatus
    cooking_time: float
    client: Client

    def __init__(self,
                 client_id: int,
                 table_id: int,
                 order_items: List[OrderItem],
                 comment: Optional[str],
                 date_time: datetime,
                 status: OrderStatus,
                 client: Client,
                 id: Optional[int] = None):
        super().__init__(
            id=id,
            client_id=client_id,
            table_id=table_id,
            order_items=order_items,
            comment=comment,
            date_time=date_time,
            status=status,
            client=client,
            cooking_time=max(item.menu_item.cooking_time_minutes for item in order_items)
        )

File: backend/Domen/Models/Order.cs
using System.Text.Json.Serialization;

namespace Domen.Models;

public class Order
{
    public int? Id { get; set; }
    [JsonPropertyName("client_id")] public int ClientId { get; set; }
    [JsonPropertyName("table_id")] public int TableId { get; set; }
    [JsonPropertyName("cooking_time")] public double CookingTime { get; set; }
    public string? Comment { get; set; }
    [JsonPropertyName("date_time")] public DateTime DateTime { get; set; }
    public OrderStatus Status { get; set; }
    public Client Client { get; set; }
    [JsonPropertyName("order_items")] public List<OrderItem> OrderItems { get; set; }

    public Order(int? id, int clientId, int tableId, string? comment, DateTime dateTime, Client client, List<OrderItem> orderItems)
    {
        Id = id;
        ClientId = clientId;
        TableId = tableId;
        Comment = comment;
        DateTime = dateTime;
        Status = OrderStatus.InAssembly;
        Client = client;
        CookingTime = orderItems.Max(item => item.MenuItem.CookingTimeMinutes);
        OrderItems = orderItems;
    }
}

public enum OrderStatus
{
    InAssembly,
    Cooking,
    Done,
}